### ISSUE TITLE
[java] UnusedPrivateField false-positive with @FXML

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#907]
**Affects PMD Version:** 6.0.1

**Rule:**
category/java/bestpractices.xml/UnusedPrivateField

**Description:**
Fields, that are annotated with `@FXML` should not be considered by the rule `UnusedPrivateField`. Removing such fields could break the JavaFX application, since the FXML and Java code need
to be in sync.

Maybe we can add a property with a list of annotations (class types), that - if present - shutdown this rule.

**Code Sample demonstrating the issue:**

```
public class NodeInfoPanelController {
    @FXML
    private TabPane nodeInfoTabPane;
}
```

**Running PMD through:** *[Maven]*

**Comment by adangel** [comment]
We probably have additional `@Deprecated` as a FP example.
